docs/custom_build.md

```diff
@@ -102,23 +102,44 @@ There are a couple of caveats you should be aware of with `buildah` (and similar
 
 All the commands above contain `$EXPECTED_REF`. What is that?
 
-### Tags
+### Build Tags
 
-Tilt always deploys with a digest, not a bare ref. (Instead of `gcr.io/company-name/frontend`, Tilt injects `gcr.io/company-name/frontend:tilt-ffd9c2013b5bf5d4`). Before explaining why (at the bottom of this document), let's describe what this means for your Tiltfile and build script.
+Tilt always deploys with a content-based, immutable tag, not a bare ref. (Instead of `gcr.io/company-name/frontend`, Tilt injects `gcr.io/company-name/frontend:tilt-ffd9c2013b5bf5d4`, where the `ffd9c2013b5bf5d4` part is based on the contents of your image). Before explaining why (see [below](#why-tilt-uses-immutable-tags)), let's describe what this means for your Tiltfile and build script.
 
-Because different build tools have different ergonomics, Tilt supports two modes:
+Because different build tools have different ergonomics, Tilt supports two modes for tagging an image in a custom_build:
 * one-time tags via $EXPECTED_REF
 * temporary refs
 
-### One-time Tags
+### One-time Build Tags
+
 This mode is easy if your tool takes the destination of the image as an argument (e.g. `docker build`).
 * Tiltfile sets a custom build command (e.g. `custom_build(..., 'docker build -t $EXPECTED_REF frontend')`).
 * Tilt sets a one-time ref as environment variable before executing the custom build command. (e.g. `EXPECTED_REF=gcr.io/company-name/frontend:tilt-ffd9c2013b5bf5d4`)
 * The custom build command builds the image and pushes to that ref (e.g. by reading `$EXPECTED_REF`).
 
-### Temporary Refs
+### Temporary Build Refs
+
 Other tools want to have an image ref hard-coded in configuration. They'll build and push to the same place each time. Instead of having to change your tool, tell Tilt what tag the build image will have with `custom_build(..., tag='frontend:s2i')`. After Tilt runs your build command, it will find this image and retag and push it with a unique ID.
 
+### Deploy Tags
+
+In rare cases, another script in your build system may need to know what tag Tilt is going to deploy.
+
+Tilt has a special command to help with this. After you build the image, run:
```

Comment:
IS there a comprehensible example you could give here? i think it would help me understand what cases this IS for so I wouldn't be tempted to use it in the wrong case

Reply:
added a note at the top of the section